src/mame/atari/atarigt.cpp

```diff
@@ -763,6 +763,21 @@ static INPUT_PORTS_START( primrage )
 	PORT_BIT( 0x08000000, IP_ACTIVE_LOW, IPT_BUTTON4 ) PORT_PLAYER(1)
 INPUT_PORTS_END
 
+static INPUT_PORTS_START( primrage )
+	PORT_INCLUDE( common )
+
+	/* primrage has a dedicated start button */
+	PORT_MODIFY( "P1_P2" )
+	PORT_BIT( 0x000000F5, IP_ACTIVE_LOW, IPT_UNUSED )
+	PORT_BIT( 0x00000008, IP_ACTIVE_LOW, IPT_BUTTON4 ) PORT_PLAYER(2)
+	PORT_BIT( 0x00000200, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_PLAYER(2)
+	PORT_BIT( 0x00000400, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_PLAYER(2)
+	PORT_BIT( 0x00000800, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_PLAYER(2)
+	PORT_BIT( 0x00000002, IP_ACTIVE_LOW, IPT_BUTTON4 ) PORT_PLAYER(1)
+	PORT_BIT( 0x02000000, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_PLAYER(1)
+	PORT_BIT( 0x04000000, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_PLAYER(1)
+	PORT_BIT( 0x08000000, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_PLAYER(1)
```

Comment:
You only need to modify the bits that are actually different.  The start buttons and buttons 1-3 are the same as in `common` – you only need to put the button 4 inputs here.  Also, please keep the inputs in ascending bit order like the rest of the file.

This should come before the `primrageo` inputs to match the ordering of the systems, too.

Reply:
The button reordering is intentional to try and keep the meanings the same as the old ROMs.  Every button has "shifted" down a number.  IE:  Old Button1 is now just start, old Button2 is now Button1, etc.  The new button mappings (0x2 and 0x8) are added as Button4.     This change in mapping can be see via the input tests in the service menu.    

Let me know if preserving this ordering is incorrect and I can switch the modification block to just do the minimal button additions, but it will mean the mame button numbers will have different meanings across rom revisions.

I'll reorder the code blocks as you suggest.

Reply:
Do you want me to break out the unused block to fully order the bits, as well?  Or something else?

e.g.

```
PORT_BIT( 0x00000001, IP_ACTIVE_LOW, IPT_UNUSED )
	PORT_BIT( 0x00000002, IP_ACTIVE_LOW, IPT_BUTTON4 ) PORT_PLAYER(1)
	PORT_BIT( 0x00000004, IP_ACTIVE_LOW, IPT_UNUSED )	
	PORT_BIT( 0x00000008, IP_ACTIVE_LOW, IPT_BUTTON4 ) PORT_PLAYER(2)
	PORT_BIT( 0x000000F0, IP_ACTIVE_LOW, IPT_UNUSED )

```